tests/net/dhcpv4/client/src/main.c

```diff
@@ -137,18 +142,28 @@ static const unsigned char ack[382] = {
 0x74, 0x65, 0x6c, 0x03, 0x63, 0x6f, 0x6d, 0x00,
 0x05, 0x69, 0x6e, 0x74, 0x65, 0x6c, 0x03, 0x63,
 0x6f, 0x6d, 0x00, 0x2c, 0x08, 0xa3, 0x21, 0x07,
-0x56, 0x8f, 0xb6, 0xfa, 0x69, 0xff
+0x56, 0x8f, 0xb6, 0xfa, 0x69, 0x2b, 0x0a, 0x01,
+0x07, 0x73, 0x74, 0x72, 0x69, 0x6e, 0x67, 0x00,
+0xff, 0x2b, 0x0f, 0x02, 0x01, 0x01, 0x03, 0x00,
+0xfe, 0x10, 0xfe, 0xfe, 0xfe, 0xfe, 0xfe, 0xfe,
+0xfe, 0xfe, 0x2b, 0x01, 0xfe, 0x46, 0x04, 0xc6,
+0x33, 0x64, 0x10, 0xff
 };
 
```

Comment:
That depends on what you mean by "invalid data in packets".
In this implementation, we have the following (total length refers to the option 43 length):
1. a properly encapsulated option 43, which contains the string "string" inside suboption 1
2. an encapsulated option with a total length of 15 with:
  a) properly encapsulated suboption 2 with a length of 1 and a value of 1 inside
  b) properly encapsulated suboption 3 with a length of 0 (i.e. no data)
  c) inproperly encapsulated suboption 254 with a length of 16, followed by 8 bytes of 254
3. an inproperly encapsulated option 43 with a length of 1 (i.e. too short to contain an encapsulated option) with 254 inside
4. option 70 with IP 198.51.100.16 (to make sure that processing is properly resumed)

Apart from checking the values of the correctly encapsulated options, there are callbacks registered for both option 254 and 43.254 that would cause an immediate failure of the test.

Reply:
Ok, it was not clear from the tests what the data is. And one would need some rfc reading to decipher the packet data to understand what is being tested. Perhaps it would make sense to describe in the test the steps you have outlined above.

Reply:
This should be addressed now. However, while doing so, I found what feels like a rather significant issue with DHCP tests:
Both arrays hold DHCP ACK messages. The one in `ack` contains no client IP and they differ in relay agent IP. I don't think this is intentional.

Reply:
@maass-hamburg please check if there is any part of the code that overwrites some bytes to turn the offer into an actual offer. If not, use a wireshark dump of some actual interaction with a DHCP server to check what the expected behavior is (I'm not even sure the options are usually sent for both ack and offer). Since sending DHCP option 60 to the server will require slightly extended tests, it would make sense to fix the current test before starting work on option 60.

Reply:
all dhcp options in the dhcp packet data are now described by comments to make them easier to understand.